UM-USElab/gradecraft-development
# Remove mongo

### Status
**READY FOR REVIEW**

### Description
This PR removes Mongo and Resque and installs Sidekiq to handle background jobs. 

### Todos

### Deploy Notes
To get this to pass Codeship, we'll need to remove the references to Resque and Mongo in the build settings. To take advantage of the cost savings possible, we'll need to remove Mongo from the setup in OpenShift.

### Gem dependencies
- ADDED Sidekiq
- REMOVED Mongo, Resque, resque-scheduler

### Migrations
NO

### Steps to Test or Reproduce
LOCALLY:
Checkout branch, restart server, confirm that a grade awarded to a student 1) updates their total course score and 2) a job is visible as successfully processing at http://localhost:5000/sidekiq

STAGING: 
Need to do QA testing on every background job

### Impacted Areas in Application
List general components of the application that this PR will affect:

* Analytics tracking
* Exporters
* Background jobs (score updating, predictor logging)

======================
Closes #4098
